src/DataStructures/Matrix.cpp

```diff
@@ -4,16 +4,55 @@
 #include "Matrix.hpp"
 
 #include <algorithm>
+#include <memory>
 #include <ostream>
 #include <pup.h>
 #include <pup_stl.h>
 
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Blas.hpp"
+
 Matrix::Matrix(const size_t number_of_rows, const size_t number_of_colums,
-               const double value)
+               const double value) noexcept
     : number_of_rows_(number_of_rows),
       number_of_columns_(number_of_colums),
       data_(number_of_rows_ * number_of_columns_, value) {}
 
+Matrix::Matrix(const std::vector<std::vector<double>>& rows) {
```

Comment:
The implementation here was updated over in #543.  That version looks good, so just copy it into this PR.